ssl/statem/statem_lib.c

```diff
@@ -1039,14 +1039,25 @@ WORK_STATE tls_finish_handshake(SSL *s, WORK_STATE wst, int clearbufs, int stop)
     int cleanuphand = s->statem.cleanuphand;
 
     if (clearbufs) {
-        if (!SSL_IS_DTLS(s)) {
+        if (!SSL_IS_DTLS(s)
+#ifndef OPENSSL_NO_SCTP
+                /*
+		 * RFC6083: SCTP provides a reliable and in-sequence transport service for DTLS
+                 * messages that require it. Therefore, DTLS procedures for retransmissions
+		 * MUST NOT be used.
+                 * Hence the init_buf can be cleared when DTLS over SCTP as transport is used.
+                 */
```

Comment:
Looks like there is mixed whitespace in lines 1044-1049?

Reply:
I would like to see this description (possibly in an abbreviated form) in the body of the commit message, too.

Reply:
I have worked on your comments, please review.